Source/WebCore/page/UserContentURLPattern.cpp

```diff
@@ -61,71 +98,92 @@ bool UserContentURLPattern::matchesPatterns(const URL& url, const Vector<String>
     return matchesAllowlist && !matchesBlocklist;
 }
 
-bool UserContentURLPattern::parse(StringView pattern)
+void UserContentURLPattern::normalizeHostAndSetMatchSubdomains()
+{
+    ASSERT(!m_matchSubdomains);
```

Comment:
It might be nice if `m_matchSubdomains` had some third state that indicated it hadn't been configured yet, then this assert would be a little more actionable.

Really, you're just trying to assert this method doesn't get called more than once, correct?

Reply:
Correct, this ASSERT is just to avoid setting `m_matchSubdomains` twice in the method, since it will start as false for a fresh object.